uberfire-provisioning-services-impl/src/main/java/org/uberfire/provisioning/services/endpoint/impl/RuntimeProvisioningServiceImpl.java

```diff
@@ -46,36 +40,26 @@
 @ApplicationScoped
 public class RuntimeProvisioningServiceImpl implements RuntimeProvisioningService {
 
-    @Context
-    private SecurityContext context;
-
     @Inject
-    private BeanManager beanManager;
-    
+    private Instance<ProviderType> providerTypeInstances;
+
     @Inject
     private RuntimeRegistry registry;
 
     private boolean initialized = false;
 
-    public RuntimeProvisioningServiceImpl() {
-
-    }
-
     @PostConstruct
     public void cacheBeans() {
         if ( !initialized ) {
-            Set<Bean<?>> beans = beanManager.getBeans(ProviderType.class, new AnnotationLiteral<Any>() {});
-            for(Bean b : beans ){
+            providerTypeInstances.iterator().forEachRemaining( pt -> {
                 try {
-                    // I don't want to register the CDI proxy, I need a fresh instance :(
-                    registry.registerProviderType((ProviderType) b.getBeanClass().newInstance());
-                } catch (InstantiationException | IllegalAccessException ex) {
-                    Logger.getLogger(RuntimeProvisioningServiceImpl.class.getName()).log(Level.SEVERE, null, ex);
+                    registry.registerProviderType( pt );
```

Comment:
I've changed this one, because if we use the Instance<ProviderType> providerTypeInstances; when Jackson do the marshalling it returns the marshalled CDI proxy. For that reason I changed to the freshly created instance. 
Can you revert this? With this code the remote rest client doesn't work